front-end-tools/src/Main.tsx

```diff
@@ -283,17 +284,34 @@ export default function Main(props: ConnectionProps) {
                 .getModuleSource(new ModuleReference(moduleReferenceCalculated))
                 .then((value) => {
                     if (value === undefined) {
-                        setModuleReferenceAlreadyDeployed(false);
+                        setIsModuleReferenceAlreadyDeployedStep1(false);
                     } else {
-                        setModuleReferenceAlreadyDeployed(true);
+                        setIsModuleReferenceAlreadyDeployedStep1(true);
                     }
                 })
                 .catch(() => {
-                    setModuleReferenceAlreadyDeployed(false);
+                    setIsModuleReferenceAlreadyDeployedStep1(false);
                 });
         }
     }, [connection, account, client, moduleReferenceCalculated]);
 
+    useEffect(() => {
```

Comment:
Isn’t it possible only to have one `moduleReferenceAlreadyDeployed` state, which could have a effect changing listening to both `moduleReferenceCalculated`and `moduleReference `? 

The warning should be shown either way regardless of where the user provides the module (from file or string).

Reply:
So how should we set `moduleReferenceAlreadyDeployed` if `moduleReferenceCalculatedand is NOT deployed` but `moduleReference is deployed` or vice versa?